### ISSUE TITLE
The return type is not correct for getProducts function

### ISSUE THREAD

**Issue by johnolofsson** [dooboolab/react-native-iap#2027]
> Please use the [Discussion board](https://github.com/dooboolab/react-native-iap/discussions) if you want to get some help. Please use issues to report bugs.

**Description**

The return type is not correct for `getProducts` function.

I'm migrating from version 5.1.1 to 11.0.7. I have read all the release notes and migrated breaking changes and it seems to work pretty well except for one thing.

For version 5.1.1 when using an Android device and I call `getProducts` I get a response like this:

```
{
  "currency": "USD",
  "description": "Some description",
  "freeTrialPeriodAndroid": "",
  "iconUrl": "",
  "introductoryPrice": "",
  "introductoryPriceCyclesAndroid": "0",
  "introductoryPricePeriodAndroid": "",
  "localizedPrice": "$13.99",
  "originalJson": {...},
  "originalPrice": "13.99",
  "originalPriceAndroid": "$13.99",
  "packageNameAndroid": "",
  "price": "13.99",
  "productId": "product-id",
  "subscriptionPeriodAndroid": "",
  "title": "This is the title",
  "type": "inapp",
  "typeAndroid": "inapp"
}
```
For version 11.0.7 I get a response like this:
```
{
  "description": "Some description",
  "name": "A name",
  "oneTimePurchaseOfferDetails": {
    "formattedPrice": "$13.99",
    "priceAmountMicros": "13990000",
    "priceCurrencyCode": "USD"
  },
  "productId": "product-id",
  "productType": "inapp",
  "title": "The title"
}
```

When reading the source code I can see that from [9.0.0 the implementation seems to have changed in the android module](https://github.com/dooboolab/react-native-iap/blob/9.0.0/android/src/play/java/com/dooboolab/RNIap/RNIapModule.kt).

Looks very much like the current response I get after migrating.

I can also see that there is a new optional field `oneTimePurchaseOfferDetails` on the `Product` type that `getProducts` is returning (well, it returns an array of products).

Now to the issue.

The `Product` type still says that `localizedPrice` for example is **non optional**. So when I migrated there is no indication that things will break for me since I relied on that the type would reflect the data I get in return.

**Expected Behavior**

When I write a (naive) function like this:
```
interface MyProductViewModel {
  priceToDisplay: string; // Required
}
async function someProduct(): Promise<MyProductViewModel> {
  const products = await getProducts({ skus: ['id-1', 'id-2'] });
  const firstProduct = products[0];

  return {
    priceToDisplay: firstProduct.localizedPrice,
  };
}
```

I would expect to get a type error on ` priceToDisplay: firstProduct.localizedPrice` since I say that `priceToDisplay` is always required but in fact `localizedPrice` is not always defined. Only for iOS devices that seems to be the case, not for Android.

I realise that the solution is to write something like `priceToDisplay: firstProduct.oneTimePurchaseOfferDetails?.formattedPrice ?? firstProduct.localizedPrice ?? 'unknown'` but the types should have helped me better here.

**Environment:**

- react-native-iap: 11.0.7
- react-native: 0.68.3
- Platforms (iOS, Android, emulator, simulator, device): Android

**To Reproduce**

1. Go to your IDE
2. Write the code snippet above
3. Error should be shown on line `priceToDisplay: firstProduct.localizedPrice`
4. Or do the same with the Typescript compiler or similar

---

[Optional] **Additional Context**

**Comment by andresesfm** [comment]
It has been fixed in [12.0.2](https://github.com/dooboolab/react-native-iap/commit/c0a0f3d0c80c8ab6378838d5b033dbbe7ebd6627)

**Comment by pierroo** [comment]
Unfortunately I would like to re-open @andresesfm , it seems that localizedPrice has disappeared again in latest version 12.4.9, are you aware of it?

**Comment by andresesfm** [comment]
Is this for StoreKit 1 @pierroo? This is where it's being passed from native: https://github.com/dooboolab/react-native-iap/blob/main/ios/RNIapIos.swift#L687

**Comment by pierroo** [comment]
I'm not familiar with the wording, I'm using the default API, but if your question is "is it storekit 2" then the answer is no. (based on your documentation)

But this localizedPrice issue is not with iOS, but on android only.
I can indeed see it as expected in the props on iOS.

**Comment by andresesfm** [comment]
Not sure why I assumed it was fixed or if this was a regresion. In any case, here is the fix to backfill those values: https://github.com/dooboolab/react-native-iap/pull/2181

**Comment by andresesfm** [comment]
Released in `12.4.11`
